ui/apps/platform/src/Containers/NetworkGraph/types/topology.type.ts

```diff
@@ -81,11 +84,52 @@ export type NamespaceData = {
 
 export type NetworkPolicyState = 'none' | 'both' | 'ingress' | 'egress';
 
+// prettier-ignore
+type NodeModelType<DataType extends NodeDataType> =
+    DataType extends 'DEPLOYMENT' ? DeploymentNodeModel :
+    DataType extends 'INTERNAL_GROUP' ? InternalGroupNodeModel :
+    DataType extends 'EXTERNAL_GROUP' ? ExternalGroupNodeModel :
+    DataType extends 'EXTERNAL_ENTITIES' ? ExternalEntitiesNodeModel :
+    DataType extends 'CIDR_BLOCK' ? CIDRBlockNodeModel :
+    DataType extends 'UKNOWN_INTERNAL_ENTITY' ? UnknownInternalEntityNodeModel :
+    DataType extends 'INTERNAL_ENTITIES' ? InternalEntitiesNodeModel :
+    DataType extends 'EXTRANEOUS' ? ExtraneousNodeModel :
+    never;
+
+/**
+ * Returns a type guard for checking if a node is of a certain type
+ *
+ * @template DataType the type of node to check, should not be specified explicitly
+ * @param type the expected string value of the node's `data.type` property
+ * @returns a type guard for checking if a node is of a certain type
+ */
+export function isOfType<DataType extends NodeDataType>(
+    type: DataType
+): (node: CustomNodeModel) => node is NodeModelType<DataType> {
+    return (node: CustomNodeModel): node is NodeModelType<DataType> => node.data.type === type;
+}
+
+/**
+ * Type guard for checking if a node is of a certain type
+ *
+ * @template DataType the type of node to check, should not be specified explicitly
+ * @param type the expected string value of the node's `data.type` property
+ * @param node the node to check
+ * @returns true if the node is of the specified type, false otherwise
+ */
+export function isNodeOfType<DataType extends NodeDataType>(
+    type: DataType,
+    node: CustomNodeModel
+): node is NodeModelType<DataType> {
+    return isOfType(type)(node);
+}
+
 export type NodeDataType =
     | 'DEPLOYMENT'
     | 'EXTERNAL_GROUP'
     | 'EXTERNAL_ENTITIES'
     | 'CIDR_BLOCK'
+    | 'INTERNAL_GROUP'
     | 'UKNOWN_INTERNAL_ENTITY'
```

Comment:
@vikin91 Should this type be the following instead? I wasn't sure if this was a typo or intentional. If the former I can go through and rebase/update this PR after the change on the backend.
```suggestion
    | 'UNKNOWN_INTERNAL_ENTITY'
```

Reply:
The `Unknown Internal Entity` was a predecessor to `Internal Entities`. It should be removed as we abandoned that idea after getting feedback from solution architects.

The idea with `Unknown Internal Entity` was to represent each unknown IP address as a separate node on the graph (named `__ip_10.x.y.z`). That lead pretty quickly to a chaos on the graph as we had too many nodes. Instead, we grouped all of those into single node 'Internal Entities`, which is our plan on how to implement right now.


The code still had the copy of `UNKNOWN_INTERNAL_ENTITY` but I disabled the mechanism in Sensor. 
I should have mentioned that, I am really sorry that I missed that in our meeting :(

Reply:
No worries - should this type be removed entirely from the PR?

Reply:
Yes please, let's remove `UNKNOWN_INTERNAL_ENTITY` and all its typoed versions ;)

Reply:
I will also remove it form Sensor and Central code to keep it clean. Just wondering if I shall do it now and you could rebase, or I would do it after we merge into my feature branch. Do you have any preference?

Reply:
I think either one should be fine. I already removed it from my branch, so I can rebase again easily if you update yours. I don't imagine there will be any conflicts to worry about.